lib/orderbook/OrderBook.ts

```diff
@@ -300,6 +300,8 @@ class OrderBook extends EventEmitter {
         portion.localId = maker.localId;
         result.internalMatches.push(maker);
         this.emit('ownOrder.filled', portion);
+        await this.repository.addOrdersIfNotExists([maker, taker]);
```

Comment:
We should add the orders from within the `persistTrade` method so we don't need to copy this line each time its called.